### DISCUSSION TITLE
Help for how to get token in a bearer token api server

### DISCUSSION THREAD

**Discussion by baalchina** [grafana/grafana-infinity-datasource#1011] [General]
Hi all, I am using infinity datasource to explorer data in my company api server, and I am confusing in  how to get the token.

For example , the url which I can get the data like this:
http://myserver/open_api/customization/full?access_token=abcdefg12345, which "abcdefg12345" is the token., which work fine with infinity.

But in fact ,the token is not given directly, I should using appkey and secret to get the token, and token is only available in a short time,like this:

1. I must post a request to server, like this:http://myserver/open_api/authentication/get_access_token?key=abcdefg&secret=123456, which abcdefg is key, and 123456 is secret.
2. Then, the server return a json response, with the access_token, expire time, and some other information.
3. I should use the access_token given in step 2, to explorer the data.
4. The token only available in a short time, after that, I shoud repeat step 1 and 2, to get a new token.

So, I am wondering, how to using infinity-datasource to get the token? 

Thanks.

**Comment by G0nz0uk** [comment]
I need a fix for this too and it seems others do to.  Did you find another way to do this at all?
